tests/models/test_default_evaluator.py

```diff
@@ -65,15 +68,91 @@ def assert_dict_equal(d1, d2, rtol):
         assert np.isclose(d1[k], d2[k], rtol=rtol)
 
 
-def test_regressor_evaluation(linear_regressor_model_uri, diabetes_dataset):
+def evaluate_model_helper(
+    model,
+    baseline_model,
+    data,
+    targets,
+    model_type: str,
+    dataset_name=None,
+    evaluators=None,
+    evaluator_config=None,
+    eval_baseline_model_only=False,
+):
+
+    """
+    Helper function for testing MLflow.evaluate
```

Comment:
Nit: Can we make this docstring a bit more perspective (i.e. what does the helper function do?)

Reply:
For testing if metrics and artifacts are not logged for evaluation of baseline model, it is hard to test that if we pass in both `model` and `baseline_model`, because it is hard to tell whether the `baseline_model` logs some metrics that overwritten the ones `model` generated, so I came up with this to test if we pass in a baseline model and only evaluate it without candidate model to verify that nothing is logged and no artifacts is generated so that we know the boolean flag`is_baseline_model` actually disable logging metrics and artifacts. To do that I have to a way to pass the boolean flag to `DefaultEvaluator` which is through the evaluator config.